### PR TITLE
feat: development of vue simulator final report

### PR DESCRIPTION
**Pull request by Arnabdaz**
- Added final report for GSoC@23 project Development of Vue Simulator.

### CODE REVIEW

**Comment by Arnabdaz** [comment]
cc @vedant-jain03 @devartstar

File: content/posts/Arnab_Das_GSoC_final_report.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+> While carrying forward the goals from last year's project, this project focuses on finishing up the development of Vue.js circuit simulator while boosting its performance, stability, and scalability by transitioning the JavaScript codebase to TypeScript. This project encompasses integrating the simulator with the main repository, carrying out new API endpoint creation and integration, introducing new features, and enhancing user experience.**
+
+### Project Goals & Accomplishments -
+
+---
+
+1. Conversion of Components to Vue.js Components
+2. Integration of Vue Simulator in the main repository
+3. Designing and implementing new API endpoints
+4. implementation of JWT for Vue Simulator
```
Comment (vedant-jain03) [inline content/posts/Arnab_Das_GSoC_final_report.md:30]:
```suggestion
4. Implementation of JWT for Vue Simulator
```

File: content/posts/Arnab_Das_GSoC_final_report.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+2. updating the user menu to make it more user-friendly and easier to add new features.
+3. updating the navbar to make it more developer friendly and user-friendly in smaller screens.
+
+**Read my [Phase 1 blog](https://blog.circuitverse.org/posts/arnab_das_gsoc_phase1/) for more details on [Implementation of new features](http://localhost:1313/posts/arnab_das_gsoc_final_report/#-implementation-of-new-features-) & [UI and UX](http://localhost:1313/posts/arnab_das_gsoc_final_report/#-enhancing-ui-and-ux-) updates**
+
+### 🔹 Performance and Stability 📈
+
+With the conversion of the simulator to Vue.js & TypeScript lots of bugs were introduced and the old methods (such as DOM manipulation for ui changes & simulator logics) were not working, for this new methods based on Vue & es6 js were used. To talk about some of the major are:
+
+1. moving the tab related data (e.g. tabs info, active tabs etc.) to be stored on Pinia Store.
```
Comment (vedant-jain03) [inline content/posts/Arnab_Das_GSoC_final_report.md:144]:
```suggestion
1. Moving the tab related data (e.g. tabs info, active tabs etc.) to be stored on Pinia Store.
```

File: content/posts/Arnab_Das_GSoC_final_report.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+9. Error in creating new circuit after deleting of circuit.
+10. Wrong Json object creation while saving circuits resulting in cashing simulator when saved circuit is loaded.
+
+These were some of the major issues fixed which hindered the Performance & stability of the Simulator, Lots of other bugs and logical changes were needed to be done for the simulator to be fully functional while being totally backward completable. For more information on these changes feel free to check out the blogs, prs & issues.
+
+### 🔹 Panels Dragging Functionality 🖱️
+
+Previously the simulator leveraged jQueryUI for the dragging functionality of the panels. However, with the transition to Vue.js, this functionality was lost.
+
+we used modern ES6 JavaScript to implement the dragging functionality. After some research and testing, we decided to use [interact.js](https://github.com/taye/interact.js). The main reason behind using this library was its small size, similar usage to jQueryUI, nearest internal implementation if it was to be implemented from scratch and most importantly it provided all the functionality needed for the simulator.
```
Comment (vedant-jain03) [inline content/posts/Arnab_Das_GSoC_final_report.md:161]:
```suggestion
We used modern ES6 JavaScript to implement the dragging functionality. After some research and testing, we decided to use [interact.js](https://github.com/taye/interact.js). The main reason behind using this library was its small size, similar usage to jQueryUI, nearest internal implementation if it was to be implemented from scratch and most importantly it provided all the functionality needed for the simulator.
```

File: content/posts/Arnab_Das_GSoC_final_report.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+{{< video src="/videos/Arnab_GSoC23/dragging.mp4" controls="true" preload="auto" >}}
+
+### 🔹 Implementation of Embed view of the simulator 🖼️
+
+Embed view provides a preview of the circuit which, in which one can't edit the circuit. this also, allows users to embed the circuit in external websites. This was implemented by creating a new route `/simualtorvue/embed/:id` which would render the embed view of the circuit.
```
Comment (vedant-jain03) [inline content/posts/Arnab_Das_GSoC_final_report.md:167]:
```suggestion
Embed view provides a preview of the circuit, in which one can't edit the circuit. This also, allows users to embed the circuit in external websites. This was implemented by creating a new route `/simualtorvue/embed/:id` which would render the embed view of the circuit.
```

File: content/posts/Arnab_Das_GSoC_final_report.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+### 🔹 Refactoring of Styles and Codebase 🔄
+
+Old simulator code was using SCSS at a lot of part of the codebase , so those were to be converted to CSS & the new components corresponding styles were moved to the corresponding Vue components `<style></style>` tags. This was done to make the codebase more developer friendly and easier to understand. Also, the clearing & refactoring of codebase was started to remove unused code and make the codebase more readable. the following folder structure was decided to be followed for the refactoring:
+
+![refactoring folder structure](/images/Arnab_GSoC23/refactor.png)
+
+### 🔹 Future Work 📅
+
+- currently the simulator is working totally fine only few functionalities have some bugs , so most important point would be fixing those to make the simulator fully ready to replace the current one.
```
Comment (vedant-jain03) [inline content/posts/Arnab_Das_GSoC_final_report.md:191]:
```suggestion
- Currently, the simulator is working totally fine only a few functionalities have some bugs, so the most important point would be fixing those to make the simulator fully ready to replace the current one.
```

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

**Review by vedant-jain03 — APPROVED** [review]
Merging this, @Arnabdaz Great work, and thankyou for your contribution. 🎉 🚀 

cc @devartstar @tachyons

**Pull request merged**